src/server/game/Entities/Player/Player.cpp

```diff
@@ -23968,8 +23968,11 @@ void Player::SetViewpoint(WorldObject* target, bool apply)
         // farsight dynobj or puppet may be very far away
         UpdateVisibilityOf(target);
 
-        if (target->isType(TYPEMASK_UNIT) && target != GetVehicleBase())
+        if (target->isType(TYPEMASK_SEER) && target != GetVehicleBase())
```

Comment:
Will do, I've pushed a new commit so that `SetViewpoint` will only call `SetSeer` and `AddPlayerToVision` for objects that specifically have `TYPEMASK_UNIT` or `TYPEMASK_PLAYER`. I also updated the "removed seer" case so that `RemovePlayerFromVision` is 1-to-1 with the number of times that `AddPlayerToVision` is called.